### PR TITLE
Issue 46

### PR DESCRIPTION
**Pull request by korikuzma**

### CODE REVIEW

File: uta_tools/data_sources/seqrepo_access.py [hunk 1]
```diff
@@ -1,68 +1,171 @@
...
+        """
+        sequence, warnings = self.check_sequence(
+            ac, start, end, residue_mode)
+        if warnings:
+            return None, warnings
+        else:
+            return sequence, None
 
+    def check_sequence(
+            self, ac: str, start: int = None,
```
Comment (jsstevenson) [inline uta_tools/data_sources/seqrepo_access.py:69]: I believe the default value for `start` will cause a valueerror on line 34

**Review by jsstevenson — CHANGES_REQUESTED** [review]

File: uta_tools/data_sources/seqrepo_access.py (hunk 1 above)
Comment (jsstevenson) [reply uta_tools/data_sources/seqrepo_access.py:69]: er TypeError

Comment (korikuzma) [reply uta_tools/data_sources/seqrepo_access.py:69]: Ope good catch. I don't think I meant to make it optional

File: uta_tools/data_sources/seqrepo_access.py [hunk 2]
```diff
@@ -1,68 +1,171 @@
...
+            # the sequence from valid positions. So we want to make sure
+            # that both start and end positions are valid
+            expected_len_of_seq = end - start
+            if len(sequence) != expected_len_of_seq:
+                return None, f"End inter-residue coordinate ({end})" \
+                             f" is out of index on {ac}"
+            return sequence, None
+
+    def is_valid_input_sequence(
+            self, ac: str, start: int = None,
```
[the remark this answers is not in the document]
Comment (korikuzma) [reply uta_tools/data_sources/seqrepo_access.py:120]: Ya i think this was a copy/paste error

**Review by jsstevenson — APPROVED** [review]

**Pull request merged**